clients/callback_client.go

```diff
@@ -157,3 +176,52 @@ func (ts TranscodeStatus) String() string {
 	}
 	return "unknown"
 }
+
+// The various status messages we can send
+
+type TranscodeStatusMessage struct {
+	CompletionRatio float64 `json:"completion_ratio"` // No omitempty or we lose this for 0% completion case
+	Error           string  `json:"error,omitempty"`
+	Retriable       *bool   `json:"retriable,omitempty"` // Has to be a pointer or we can't differentiate omission from 'false'
+	Status          string  `json:"status,omitempty"`
+	Timestamp       int64   `json:"timestamp"`
+}
+
+type InputVideoTrack struct {
+	Fps         int     `json:"fps,omitempty"`
+	Type        string  `json:"type,omitempty"`
+	Codec       string  `json:"codec,omitempty"`
+	Width       int     `json:"width,omitempty"`
+	Height      int     `json:"height,omitempty"`
+	Bitrate     int     `json:"bitrate,omitempty"`
+	Duration    float64 `json:"duration,omitempty"`
+	PixelFormat string  `json:"pixel_format,omitempty"`
+	Channels    int     `json:"channels,omitempty"`
+	SampleRate  int     `json:"sampleRate,omitempty"`
+	SizeBytes   int     `json:"size,omitempty"`
+}
+
+type InputVideo struct {
```

Comment:
Could we get a total file size as well? (Does it make sense to have it in the tracks? I think only in the whole file is sufficient)

Reply:
Added! I think just on the input video as a whole makes sense